cmd/api/main.go

```diff
@@ -87,6 +89,21 @@ func NewServer(k8sClient kubernetes.Interface, controller routers.Controller, ca
 	router.GET("/livez", controller.Livez)
 	router.GET("/readyz", controller.Readyz)
 
+	if os.Getenv(enablePprofEnvVar) == "true" {
+		router.GET("/debug/pprof/", gin.WrapF(pprof.Index))
+		router.GET("/debug/pprof/cmdline", gin.WrapF(pprof.Cmdline))
+		router.GET("/debug/pprof/profile", gin.WrapF(pprof.Profile))
+		router.POST("/debug/pprof/symbol", gin.WrapF(pprof.Symbol))
+		router.GET("/debug/pprof/symbol", gin.WrapF(pprof.Symbol))
+		router.GET("/debug/pprof/trace", gin.WrapF(pprof.Trace))
+		router.GET("/debug/pprof/allocs", gin.WrapH(pprof.Handler("allocs")))
+		router.GET("/debug/pprof/block", gin.WrapH(pprof.Handler("block")))
+		router.GET("/debug/pprof/goroutine", gin.WrapH(pprof.Handler("goroutine")))
+		router.GET("/debug/pprof/heap", gin.WrapH(pprof.Handler("heap")))
+		router.GET("/debug/pprof/mutex", gin.WrapH(pprof.Handler("mutex")))
+		router.GET("/debug/pprof/threadcreate", gin.WrapH(pprof.Handler("threadcreate")))
+	}
+
```

Comment:
since this is also done in the sink do you think it would make sense to put in this function that would accept router as a param and both the sink and the api could import it?

something like

```
func SetupPprof(router *gin.Engine) {
	...
}
```

Reply:
I changed it to be a function and moved it under the `observability` package on `pkg`. Let me know how it looks.